wadm/lib/wadm/model/store.ex

```diff
@@ -0,0 +1,222 @@
+defmodule Wadm.Model.Store do
+  require Logger
+
+  defmodule ModelVersion do
+    @moduledoc """
+    Represents a model version and its associated creation timestamp as an RFC 3339 time
+    """
+    @type t :: %__MODULE__{
+            name: String.t(),
+            version: String.t(),
+            created: String.t(),
+            description: String.t() | nil
+          }
+    @derive Jason.Encoder
+    defstruct [:name, :version, :created, :description]
+  end
+
+  defmodule ModelMap do
+    @moduledoc """
+    A struct that contains the original, unaltered submitted text for an application specification
+    alongside its normalized and validated spec
+    """
+    @type t :: %__MODULE__{
+            raw: binary(),
+            vetted: Wadm.Model.AppSpec.t()
+          }
+
+    @derive Jason.Encoder
+    defstruct [:raw, :vetted]
+  end
+
+  @type model_map :: Map.t()
+
+  def lattice_models(redis, lattice_id) when is_binary(lattice_id) do
```

Comment:
This could use a doc indicating it returns the most recent version for each model